control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -391,39 +407,212 @@ func (r Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkainterna
+
+		// if the event was wrapped inside an error, consider the reconciliation as failed
+		if _, isEvent := err.(*reconciler.ReconcilerEvent); !isEvent {
+			return err
+		}
+	}
+
+	err = r.reconcileScaledObject(ctx, scaledObject, cg)
+
+	// if the event was wrapped inside an error, consider the reconciliation as failed
+	if _, isEvent := err.(*reconciler.ReconcilerEvent); !isEvent {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) retrieveSaslTypeIfPresent(ctx context.Context, cg *kafkainternals.ConsumerGroup) (*string, error) {
+	if hasNetSpecAuthConfig(cg.Spec.Template.Spec.Auth) && cg.Spec.Template.Spec.Auth.NetSpec.SASL.Enable {
+		if cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef != nil {
+			secretKeyRefName := cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef.Name
+			secretKeyRefKey := cg.Spec.Template.Spec.Auth.NetSpec.SASL.Type.SecretKeyRef.Key
+			secret, err := r.KubeClient.CoreV1().Secrets(cg.Namespace).Get(ctx, secretKeyRefName, metav1.GetOptions{})
+			if err != nil {
+				return nil, reconciler.NewEvent(corev1.EventTypeWarning, "SaslTypeSecretUnavailable", "Unable to get SASL type from secret: \"%s/%s\", %w", cg.Namespace, secretKeyRefName, err)
+			}
+			saslTypeValue := string(secret.Data[secretKeyRefKey])
+			return &saslTypeValue, nil
+		}
+	}
+	return nil, nil
+}
+
+func (r *Reconciler) reconcileScaledObject(ctx context.Context, expectedScaledObject *kedav1alpha1.ScaledObject, obj metav1.Object) error {
+	scaledObject, err := r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Get(ctx, expectedScaledObject.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		scaledObject, err = r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Create(ctx, expectedScaledObject, metav1.CreateOptions{})
+		if err != nil {
+			return kedafunc.ScaleObjectFailed(expectedScaledObject.Namespace, expectedScaledObject.Name, err)
+		}
+		return kedafunc.ScaleObjectCreated(scaledObject.Namespace, scaledObject.Name)
+	} else if err != nil {
+		logging.FromContext(ctx).Errorw("Unable to get an existing ScaledObject", zap.Error(err))
+		return err
+	} else if !metav1.IsControlledBy(scaledObject, obj) {
+		return fmt.Errorf("ScaledObject %q is not owned by %q", scaledObject.Name, obj)
+	} else if !equality.Semantic.DeepDerivative(scaledObject.Spec, expectedScaledObject.Spec) {
+		logging.FromContext(ctx).Debug(fmt.Sprintf("ScaledObject changed, found: %#v expected: %#v", scaledObject.Spec, expectedScaledObject.Spec))
+		scaledObject.Spec = expectedScaledObject.Spec
+		if _, err = r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Update(ctx, scaledObject, metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+		return kedafunc.ScaleObjectUpdated(scaledObject.Namespace, scaledObject.Name)
+	} else {
+		logging.FromContext(ctx).Debugw("Reusing existing ScaledObject", zap.Any("ScaledObject", scaledObject))
+	}
```

Comment:
```suggestion
	scaledObject, err := r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Get(ctx, expectedScaledObject.Name, metav1.GetOptions{})
	if error != nil && !apierrors.IsNotFound(err) {
	  return fmt.Errorf("failed to get scaledobject %s/%s: %w", expectedScaledObject.Namespace, expectedScaledObject.Name, err)
	}
	if apierrors.IsNotFound(err) {
		scaledObject, err = r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Create(ctx, expectedScaledObject, metav1.CreateOptions{})
		if err != nil {
			return fmt.Errorf("failed to create scaledobject %s/%s: %w", expectedScaledObject.Namespace, expectedScaledObject.Name, err)
		}
		// Emit ScaledObject created event
		return nil
	} 
	if !metav1.IsControlledBy(scaledObject, obj) {
		return fmt.Errorf("ScaledObject %s/%s is not owned by %s/%s", expectedScaledObject.Namespace, expectedScaledObject.Name, obj.GetNamespace(), obj.GetName())
	} 
	if !equality.Semantic.DeepDerivative(expectedScaledObject.Spec, scaledObject.Spec) { // Note expected should be the first parameter
		if _, err = r.KedaClient.KedaV1alpha1().ScaledObjects(expectedScaledObject.Namespace).Update(ctx, expectedScaledObject, metav1.UpdateOptions{}); err != nil {
			return fmt.Errorf("failed to update scaled object %s/%s: %w", expectedScaledObject.Namespace, expectedScaledObject.Name)
		}
		// Emit scaled object updated event
	}
```

Reply:
Similar applies to other functions `reconcileTriggerAuthentication`, `reconcileSecret`, etc